src/EditorFeatures/CSharpTest2/Recommendations/ThisKeywordRecommenderTests.cs

```diff
@@ -632,5 +632,71 @@ void Main(string[] args)
 }
 ", matchPriority: SymbolMatchPriority.Keyword);
         }
+
+        [Fact]
+        public async Task TestExtensionMethods_FirstParameter_AfterRefKeyword()
+        {
+            await VerifyKeywordAsync(@"
+public static class Extensions
+{
+    public static void Extension(ref $$ object obj, int x) { }
+}
+");
+        }
+
+        [Fact]
+        public async Task TestExtensionMethods_FirstParameter_AfterRefReadOnlyKeywords()
+        {
+            await VerifyKeywordAsync(@"
+public static class Extensions
+{
+    public static void Extension(ref readonly $$ object obj, int x) { }
+}
+");
+        }
+
+        [Fact]
+        public async Task TestExtensionMethods_FirstParameter_AfterInKeyword()
+        {
+            await VerifyKeywordAsync(@"
+public static class Extensions
+{
+    public static void Extension(in $$ object obj, int x) { }
```

Comment:
'in' is contextual.  so does this work if the code is just "public static void Extension(in $$"?